core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/logreplay/storage/ReplayLogStorage.scala

```diff
@@ -90,38 +93,19 @@ object ReplayLogStorage {
     }
   }
 
-  def getLogStorage(storageType: String, name: String): ReplayLogStorage = {
-    storageType match {
-      case "local" => new LocalFSLogStorage(name)
-      case "hdfs" =>
-        val hdfsIP: String =
-          AmberConfig.faultToleranceLogStorage
-        new HDFSLogStorage(name, hdfsIP)
-      case "none" =>
-        new EmptyLogStorage()
-      case other => throw new RuntimeException("Cannot support log storage type of " + other)
+  def getLogStorage(logFolderURI: Option[URI]): ReplayLogStorage = {
+    logFolderURI match {
+      case Some(value) => new URILogStorage(value)
+      case None        => new EmptyLogStorage()
     }
   }
 }
 
 abstract class ReplayLogStorage {
 
-  def getWriter: ReplayLogWriter
+  def getWriter(logFileName: String): ReplayLogWriter
 
-  def getReader: ReplayLogReader
-
-  def isLogAvailableForRead: Boolean
-
-  def deleteLog(): Unit
-
-  def cleanPartiallyWrittenLogFile(): Unit
-
-  protected def copyReadableLogRecords(writer: ReplayLogWriter): Unit = {
-    val recordIterator = getReader.mkLogRecordIterator()
-    while (recordIterator.hasNext) {
-      writer.writeLogRecord(recordIterator.next())
-    }
-    writer.close()
-  }
+  def getReader(logFileName: String): ReplayLogReader
 
+  def deleteFolder(): Unit
```

Comment:
I think the folder is an "implementation detail" to some system and maybe just call it delete storage?